RecoTauTag/RecoTau/interface/DeepTauBase.h

```diff
@@ -89,17 +101,45 @@ namespace deep_tau {
     static std::unique_ptr<DeepTauCache> initializeGlobalCache(const edm::ParameterSet& cfg);
     static void globalEndJob(const DeepTauCache* cache) {}
 
+    template <typename ConsumeType>
+    struct TauDiscInfo {
+      edm::InputTag label;
+      edm::Handle<ConsumeType> handle;
+      edm::EDGetTokenT<ConsumeType> disc_token;
+      double cut;
+      void fill(const edm::Event& evt) { evt.getByToken(disc_token, handle); };
+    };
+
+    // select boolean operation on prediscriminants (and = 0x01, or = 0x00)
+    uint8_t andPrediscriminants_;
+    std::vector<TauDiscInfo<pat::PATTauDiscriminator>> patPrediscriminants_;
+    std::vector<TauDiscInfo<reco::PFTauDiscriminator>> recoPrediscriminants_;
+
+    enum BasicDiscriminator {
+      ChargedIsoPtSum,
+      NeutralIsoPtSum,
+      NeutralIsoPtSumWeight,
+      FootprintCorrection,
+      PhotonPtSumOutsideSignalCone,
+      PUcorrPtSum
+    };
+
   private:
     virtual tensorflow::Tensor getPredictions(edm::Event& event, edm::Handle<TauCollection> taus) = 0;
     virtual void createOutputs(edm::Event& event, const tensorflow::Tensor& pred, edm::Handle<TauCollection> taus);
 
   protected:
     edm::EDGetTokenT<TauCollection> tausToken_;
-    edm::EDGetTokenT<pat::PackedCandidateCollection> pfcandToken_;
+    edm::EDGetTokenT<CandidateType> pfcandToken_;
     edm::EDGetTokenT<reco::VertexCollection> vtxToken_;
     std::map<std::string, WPList> workingPoints_;
+    const bool is_online;
     OutputCollection outputs_;
     const DeepTauCache* cache_;
+
+    static const std::map<BasicDiscriminator, std::string> stringFromDiscriminator_;
+    static const std::vector<BasicDiscriminator> requiredBasicDiscriminators;
+    static const std::vector<BasicDiscriminator> requiredBasicDiscriminatorsdR03;
```

Comment:
I just realized that not all members have an underscore at the end. Could you please add that in order to stick to the naming conventions? (is_online and the two vectors)

Reply:
... and good point to move that part to the private sector!